src/_pytest/pathlib.py

```diff
@@ -707,8 +707,12 @@ def bestrelpath(directory: Path, dest: Path) -> str:
     # Can be the case for a relative path and an absolute path.
     if not base:
         return str(dest)
-    reldirectory = directory.relative_to(base)
-    reldest = dest.relative_to(base)
+    try:
+        reldirectory = directory.relative_to(base)
+        reldest = dest.relative_to(base)
+    except ValueError:
```

Comment:
Could you please add a test for `bestrelpath` which tests this condition? It is important to avoid future regressions.

Reply:
Absolutely. Just added and let's wait for the checks to finish..

Reply:
Thanks!